daphne_worker/src/dap.rs

```diff
@@ -672,10 +685,13 @@ where
                 .map_err(dap_err)?;
 
             for report_hex in reports_from_durable {
-                let report = Report::get_decoded(&hex::decode(&report_hex).map_err(|_| {
+                let dehexed = hex::decode(&report_hex).map_err(|_| {
                     DapError::fatal("response from ReportsPending is not valid hex")
-                })?)?;
-
+                })?;
+                let task_id = task_id_from_report(&dehexed)?;
```

Comment:
I was thinking about whether we can avoid this copy here, and I don't think there's a way, sadly. `get_task_config_for()` takes in a `Cow`, which let's us use a reference to an `Id` instead of an owned `Id`, but I don't know if we can coerce a slice into a `&[u8; 32]` without a copy.

Reply:
I don't know how to do it either.